### PR TITLE
Functions with no arguments

### PR DESCRIPTION
**Pull request by llllllllll**
implements: #162

### CODE REVIEW

File: datashape/util/tests/test_testing.py [hunk 1]
```diff
@@ -190,3 +190,33 @@ def test_nested():
         )
     assert "'int32' != 'float32'" in str(e.value)
     assert "_.measure['a'].measure['b'].name" in str(e.value)
+
+
+def test_function():
+    base = dshape('(int32, int64) -> int64')
+    assert_dshape_equal(base, base)
+    cases = (
```
Comment (cpcloud) [inline datashape/util/tests/test_testing.py]: any reason not to make this a `pytest.mark.parametrize` function?

File: datashape/tests/test_parser.py [hunk 2]
```diff
@@ -611,9 +624,11 @@ def test_funcproto_no_return_type(sym):
         parse('(int64, int32) ->', sym)
 
 
-def test_empty_tuple_fails(sym):
-    with pytest.raises(DataShapeSyntaxError):
-        parse('()', sym)
+def test_empty_tuple(sym):
```
Comment (cpcloud) [inline datashape/tests/test_parser.py]: Can you add a test for a function signature like this `() -> datetime`? Doesn't have to be `datetime` could be something else, just need to test a function signature with no arguments that returns a value

File: datashape/util/tests/test_testing.py (hunk 1 above)
Comment (llllllllll) [inline datashape/util/tests/test_testing.py]: I didn't want to re-test the base case case over an over. I can split it out into two functions

File: datashape/tests/test_parser.py (hunk 2 above)
Comment (llllllllll) [inline datashape/tests/test_parser.py]: there is a test for `() -> bool` in `test_funcproto`

**Comment by llllllllll** [comment]
merging on pass

**Pull request merged**